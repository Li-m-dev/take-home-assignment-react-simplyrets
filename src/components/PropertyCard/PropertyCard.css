.card {
  width: 315px;
  padding: 18px;
  overflow: hidden;
}
.property_img {
  height: 280px;
}
.img_container {
  position: relative;
}
.heart-icon {
  position: absolute;
  top: 18px;
  right: 24px;
  height: 36px;
  width: 39px;
}

.info_area {
  font-size: 20px;
  font-weight: 600;
}

.info_price {
  font-size: 26px;
  font-weight: 700;
}

.info_address {
  font-size: 15px;
  font-weight: 400;
}

.info_date {
  font-size: 14px;
  font-weight: 400;
  color: #979797;
}
