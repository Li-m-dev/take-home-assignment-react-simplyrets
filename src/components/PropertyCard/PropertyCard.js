import React, { useState } from 'react';
import './PropertyCard.css';
import { getFormattedPrice } from '../../utils/get-formatted-price';
import { getFormattedDate } from 'utils/get-formatted-date';
import { getBathCount } from 'utils/get-bath-count';
import heartFill from '../../assets/heart-fill.svg';
import heartStroke from '../../assets/heart-stroke.svg';

const PropertyCard = ({
  address,
  photos,
  property,
  listPrice,
  listDate,
  mlsId,
  isSavedFav,
  onToggleFav,
}) => {
  const { bedrooms, bathsFull, bathsHalf, area } = property;
  const { streetNumberText, streetName, city, state } = address;

  const [isFav, setIsFav] = useState(isSavedFav);

  const toggleFav = (id) => {
    console.log('isFav: ', isFav);
    setIsFav(!isFav);
    onToggleFav(id);
  };

  return (
    <div className="card">
      <div className="img_container">
        <img className="property_img" src={photos[0]} alt="house pic" />
        <img
          data-testid="heartIcon"
          src={isFav ? heartFill : heartStroke}
          className="heart-icon"
          alt="favorite icon"
          onClick={() => toggleFav(mlsId)}
        />
      </div>
      <div className="info">
        <div className="info_area">
          {bedrooms} BR | {getBathCount(bathsFull, bathsHalf)} Bath | {area} Sq
          Ft
        </div>
        <div className="info_price">{getFormattedPrice(listPrice)}</div>
        <div className="info_address">
          {streetNumberText} {streetName}, {city}, {state}
        </div>
        <div className="info_date">Listed: {getFormattedDate(listDate)}</div>
      </div>
    </div>
  );
};

export default PropertyCard;
