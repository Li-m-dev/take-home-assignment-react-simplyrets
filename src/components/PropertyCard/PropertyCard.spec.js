const onToggleFavMock = jest.fn();

import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import PropertyCard from './PropertyCard';
import * as PropertyCardRoot from './PropertyCard';

const mockProps = {
  address: {
    streetNumberText: '111',
    streetName: 'Santa Fe',
    city: 'Houston',
    state: 'TX',
  },
  photos: 'test',
  property: { bedrooms: 4, bathsFull: 3, bathsHalf: 1, area: 2210 },
  listPrice: 300000,
  listDate: '1994-10-25T13:58:17.284009Z',
  mlsId: 123,
  isSavedFav: false,
  onToggleFav: onToggleFavMock,
};

describe('PropertyCard tests', () => {
  it('should render property card', () => {
    // Act
    const { container } = render(<PropertyCard {...mockProps} />);

    // Assert
    expect(container).toMatchSnapshot();
  });

  it('should call toggleFav on click', () => {
    // Arrange

    // const temp = jest.spyOn(PropertyCardRoot, 'toggleFav');
    // Act
    const { container, getByTestId } = render(<PropertyCard {...mockProps} />);
    const heartIcon = getByTestId('heartIcon');
    fireEvent.click(heartIcon);

    // Assert
    expect(container).toMatchSnapshot();
    // expect(temp).toHaveBeenCalledTimes(1);
    expect(onToggleFavMock).toHaveBeenCalledTimes(1);
  });
});
