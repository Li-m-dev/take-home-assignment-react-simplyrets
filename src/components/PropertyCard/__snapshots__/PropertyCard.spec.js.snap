// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard tests should call toggleFav on click 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="img_container"
    >
      <img
        alt="house pic"
        class="property_img"
        src="t"
      />
      <img
        alt="favorite icon"
        class="heart-icon"
        data-testid="heartIcon"
        src="heart-fill.svg"
      />
    </div>
    <div
      class="info"
    >
      <div
        class="info_area"
      >
        4
         BR | 
        3.5
         Bath | 
        2210
         Sq Ft
      </div>
      <div
        class="info_price"
      >
        $300,000
      </div>
      <div
        class="info_address"
      >
        111
         
        Santa Fe
        , 
        Houston
        , 
        TX
      </div>
      <div
        class="info_date"
      >
        Listed: 
        10/25/1994
      </div>
    </div>
  </div>
</div>
`;

exports[`PropertyCard tests should render property card 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="img_container"
    >
      <img
        alt="house pic"
        class="property_img"
        src="t"
      />
      <img
        alt="favorite icon"
        class="heart-icon"
        data-testid="heartIcon"
        src="heart-stroke.svg"
      />
    </div>
    <div
      class="info"
    >
      <div
        class="info_area"
      >
        4
         BR | 
        3.5
         Bath | 
        2210
         Sq Ft
      </div>
      <div
        class="info_price"
      >
        $300,000
      </div>
      <div
        class="info_address"
      >
        111
         
        Santa Fe
        , 
        Houston
        , 
        TX
      </div>
      <div
        class="info_date"
      >
        Listed: 
        10/25/1994
      </div>
    </div>
  </div>
</div>
`;
